testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/management/cli/DeployAllDomainTestCase.java

```diff
@@ -447,4 +448,225 @@ public void testLegacyEnableSingleAppDeployment() throws CommandLineException, I
         checkExist(result, cliTestApp2War.getName(), ADDED, ctx);
         checkExist(result, cliTestAppEar.getName(), ENABLED, ctx);
     }
+
+    /**
+     * Disabling all deployed application deployments between all server groups.
+     * Separate test for check status of issue WFCORE-3562 by Aesh command.
+     * Concentration is for disabling all deployments with all group by Aesh commands.
+     *
+     * @see <a href="https://issues.jboss.org/browse/WFCORE-3562">WFCORE-3562</a>
+     */
+    @Test
+    public void testDisableAllAppDeploymentAllGroup() throws CommandLineException, IOException {
+        // Step 1) Deploy applications deployments to defined server groups
+        ctx.handle("deployment deploy-file --server-groups=" + sgOne + ' ' + cliTestApp1War.getAbsolutePath());
+        ctx.handle("deployment deploy-file --server-groups=" + sgOne + ' ' + cliTestAnotherWar.getAbsolutePath());
+        ctx.handle("deployment deploy-file --server-groups=" + sgTwo + ' ' + cliTestApp2War.getAbsolutePath());
+        ctx.handle("deployment deploy-file --server-groups=" + sgTwo + ',' + sgOne + ' ' + cliTestAppEar.getAbsolutePath());
+
+
+        // Step 2) Disabling two selected applications deployments
+        ctx.handle("deployment disable --server-groups=" + sgOne + ' ' + cliTestApp1War.getName());
+        ctx.handle("deployment disable --server-groups=" + sgTwo + ' ' + cliTestApp2War.getName());
+
+        // Step 3) Disable all deployed applications deployments in all server groups
+        ctx.handle("deployment disable-all --all-relevant-server-groups");
+
+        // Step 4) Check if all applications deployments is disabled in all server groups
+        DeploymentInfoResult result = deploymentInfo(cli, sgOne);
+        checkExist(result, cliTestApp1War.getName(), ADDED, ctx);
+        checkExist(result, cliTestAnotherWar.getName(), ADDED, ctx);
+        checkExist(result, cliTestApp2War.getName(), NOT_ADDED, ctx);
+        checkExist(result, cliTestAppEar.getName(), ADDED, ctx);
+
+        result = deploymentInfo(cli, sgTwo);
+        checkExist(result, cliTestApp1War.getName(), NOT_ADDED, ctx);
+        checkExist(result, cliTestAnotherWar.getName(), NOT_ADDED, ctx);
+        checkExist(result, cliTestApp2War.getName(), ADDED, ctx);
+        checkExist(result, cliTestAppEar.getName(), ADDED, ctx);
+    }
+
+    /**
+     * Disabling all deployed application deployments by only one server group from two server groups.
```

Comment:
"Disabling all deployed application deployments by only one server group from two server groups." Seems that your test is doing: "Disabling all deployments of one server group that are also present in a second server group."

Some comments I made in previous test also apply to the content of this test.

Reply:
Thank you, your explanation in a comment are better than mine. Replaced